libraries/AP_Motors/AP_MotorsMatrix.cpp

```diff
@@ -583,42 +583,31 @@ void AP_MotorsMatrix::setup_motors(motor_frame_class frame_class, motor_frame_ty
             }
             break;
 
-        case MOTOR_FRAME_DODECAHEXA: {
-            switch (frame_type) {
-                case MOTOR_FRAME_TYPE_PLUS:
-                    add_motor(AP_MOTORS_MOT_1,     0, AP_MOTORS_MATRIX_YAW_FACTOR_CW,   1);  // TU(Top Up)
-                    add_motor(AP_MOTORS_MOT_2,     0, AP_MOTORS_MATRIX_YAW_FACTOR_CCW,  2);  // TD(Top Down)
-                    add_motor(AP_MOTORS_MOT_3,    60, AP_MOTORS_MATRIX_YAW_FACTOR_CCW,  3);  // URU(Upper Right Up)
-                    add_motor(AP_MOTORS_MOT_4,    60, AP_MOTORS_MATRIX_YAW_FACTOR_CW,   4);  // URD(Upper Right Down)
-                    add_motor(AP_MOTORS_MOT_5,   120, AP_MOTORS_MATRIX_YAW_FACTOR_CW,   5);  // LRU(Lower Right Up)
-                    add_motor(AP_MOTORS_MOT_6,   120, AP_MOTORS_MATRIX_YAW_FACTOR_CCW,  6);  // LRD(Lower Right Down)
-                    add_motor(AP_MOTORS_MOT_7,   180, AP_MOTORS_MATRIX_YAW_FACTOR_CCW,  7);  // BU(Bottom Up)
-                    add_motor(AP_MOTORS_MOT_8,   180, AP_MOTORS_MATRIX_YAW_FACTOR_CW,   8);  // BD(Bottom Down)
-                    add_motor(AP_MOTORS_MOT_9,  -120, AP_MOTORS_MATRIX_YAW_FACTOR_CW,   9);  // LLU(Lower Left Up)
-                    add_motor(AP_MOTORS_MOT_10, -120, AP_MOTORS_MATRIX_YAW_FACTOR_CCW, 10);  // LLD(Lowoe Left Down)
-                    add_motor(AP_MOTORS_MOT_11,  -60, AP_MOTORS_MATRIX_YAW_FACTOR_CCW, 11);  // ULU(Upper Left Up)
-                    add_motor(AP_MOTORS_MOT_12,  -60, AP_MOTORS_MATRIX_YAW_FACTOR_CW,  12);  // ULD(Upper Left Down)
+        case MOTOR_FRAME_DODECAHEXA:
```

Comment:
Q1: This is probably a mistake right?
A1: YES. I did not understand that there are 8 consecutive PWMs assigned to the motor.

Q2: backing out the changes to add support for motors 7 to 12?
A2: YES. Do not control 7 to 12 motors separately. Use "Y cable" to control the two motors. For example, assume that the signals of PWM1 are M1 and M12, the signals of PWM2 are M2 and M3, the signals of PWM3 are M4 and M5, the signals of PWM4 are M6 and M7, the signals of PWM5 are M8 and M9, the signals of PWM6 are M10 and M11 By giving it, it seems that it can be logically controlled as a + (plus) HEXA copter.

TYPE X:
Motor position: odd number up, even number down
M11, M12         M1,  M2
M 9, M10    X   M3,  M4
M 7, M  8          M5, M6